<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="test">
    <!-- INCLUDES -->
    <xacro:include filename="gazebo_materials.xacro" />
    <xacro:include filename="inertial_macros.xacro" />

    <!-- PROPERTIES -->
    <xacro:property name="PI" value="3.1415926535897931"/>

    <!-- BASE LINK -->
    <link name="base_link">
    </link>

    <!-- WHEEL JOINT AND LINK -->
    <joint name="wheel_joint" type="continuous">
        <origin xyz="0 0 0.1" rpy="0 0 0" />
        <parent link="base_link" />
        <child link="wheel_link" />
        <axis xyz="0.0 0.0 1.0"/>
    </joint>

    <link name="wheel_link">
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
            <geometry>
                <box size="0.5 0.5 0.1" />
            </geometry>
            <material name="red" />
        </visual>
        <collision>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
            <geometry>
                <box size="0.5 0.5 0.1" />
            </geometry>
            <material name="red" />
        </collision>
        <xacro:inertial_box mass="1.0" x="0.5" y="0.5" z="0.1">
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
        </xacro:inertial_box>
    </link>


    <ros2_control name="Wheel" type="actuator">
        <hardware>
            <plugin>rmd_hardware_interface/RMDHardwareInterface</plugin>
            <param name="interface">can0</param>
        </hardware>
        <joint name="wheel_joint">
            <param name="node_id">1</param>
            <param name="model">RMDX8_19</param>
            <param name="control_mode">torque</param>
            <param name="kp">0.0</param>
            <param name="kd">0.0</param>
            <param name="reduction">1.0</param>
            <param name="offset">0.0</param>
            <param name="home_on_startup">False</param>
            <param name="homing_torque">0.5</param>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
            <command_interface name="effort"/>
        </joint>
    </ros2_control>
</robot>